src/model/SplineSegment.cpp

```diff
@@ -0,0 +1,76 @@
+#include "SplineSegment.h"
+
+#include <cmath>
+
+SplineSegment::SplineSegment(const Point& p, const Point& q):
+        firstKnot(p), firstControlPoint(p), secondKnot(q), secondControlPoint(q) {}
+
+SplineSegment::SplineSegment(const Point& p, const Point& fp, const Point& sp, const Point& q) {
+    firstKnot = p;
+    firstControlPoint = fp;
+    secondControlPoint = sp;
+    secondKnot = q;
+}
+
+void SplineSegment::draw(cairo_t* cr) {
+    cairo_move_to(cr, firstKnot.x, firstKnot.y);
+    cairo_curve_to(cr, firstControlPoint.x, firstControlPoint.y, secondControlPoint.x, secondControlPoint.y,
+                   secondKnot.x, secondKnot.y);
+}
+
+auto SplineSegment::toVectorOfPoints() -> std::vector<Point> {
+    std::vector<Point> pointArray = {};
+    if (isFlatEnough()) {
+        pointArray = {firstKnot, secondKnot};
+    } else {
+        std::pair<SplineSegment, SplineSegment>&& twoSegments = subdivide(0.5);
+        pointArray = std::move(twoSegments.first.toVectorOfPoints());
+        pointArray.pop_back();
+        std::vector<Point>&& secondArray = twoSegments.second.toVectorOfPoints();
+        pointArray.insert(pointArray.end(), std::make_move_iterator(secondArray.begin()),
+                          std::make_move_iterator(secondArray.end()));
+    }
+    return pointArray;
+}
```

Comment:
Generally use early return as it increases readability, also it might be more efficient here.
Use "auto const&" for long constant return values or introduce a type alias and use it as "T const&".
Try to avoid to move into return value container, as it breaks possible return value optimisation. Instead assign it to a new local variable. 
The pop_back can be optimized out if the whole function only returns n-1 points.
The move_iterator is not necessary since Points are not movable.
Also vectors have a very bad scaling when resized often better use std::list or std::deque. 

Here an example:

```suggestion
auto SplineSegment::toPointSequence() const -> std::list<Point> {
    auto result = toPointSequenceImpl();
    result.emplace_back(secondKnot);
    return result;
}

auto SplineSegment::toPointSequenceImpl() const -> std::list<Point> {
    if (isFlatEnough()) {
        return {firstKnot};
    }
    auto const& childSegments = subdivide(0.5);
    auto left_points = childSegments.first.toPointSequenceImpl();
    left_points.splice(end(left_points), childSegments.second.toPointSequenceImpl());
    return left_points;
}
```